specs/ChainSpec/Upgradability.md

```diff
@@ -0,0 +1,71 @@
+# Upgradability
+
+This part of specification describes specifics of upgrading the protocol, and touches on few different parts of the system.
+
+High level upgradability consists of three parts:
+1. Updating without any changes to underlaying data structures or protocol;
+2. Updating when underlaying data structures changed (config, database or something else internal to the node and probably client specific);
```

Comment:
The NEP will benefit in the clearness if the it will focus only on the protocol-level (consensus) upgradability and not mix it with implementations upgradability.

Reply:
It does only cover protocol level upgradability. I don't think simply laying out the possible levels of upgradability is the same as covering them.

Reply:
Cool! Just to make sure :)